addons/fetchmail/models/fetchmail.py

```diff
@@ -195,11 +196,13 @@ def fetch_mail(self):
                             except Exception:
                                 _logger.info('Failed to process mail from %s server %s.', server.server_type, server.name, exc_info=True)
                                 failed += 1
+                                failed_in_loop += 1
                             self.env.cr.commit()
-                        if num_messages < MAX_POP_MESSAGES:
+                        _logger.info("Fetched %d email(s) on %s server %s; %d succeeded, %d failed.", num, server.server_type, server.name, (num - failed), failed)
```

Comment:
Since we're logging after each chunk, I think we can stick to `num` and `failed_in_loop`. I guess the global `failed` count is not so useful anymore.
```suggestion
                        _logger.info("Fetched %d email(s) on %s server %s; %d succeeded, %d failed.", num, server.server_type, server.name, (num - failed_in_loop), failed_in_loop)
```